cmd/manager/main.go

```diff
@@ -247,6 +188,16 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "AtlasDatabaseUser")
 		os.Exit(1)
 	}
+
+	dbaasEnabled, err := checkAndEnableDBaaS(mgr, logger, config)
+	if err != nil {
```

Comment:
@redhatHameed This happens only when the operator was not able to query the existence of the CRD, or could not create a controller. This means there is a serious issue such as network or resource issues and it is better to terminate the operator pod so that it can be retried.

Reply:
@jianrongzhang89 normally i'd agree with you, but since the dbaas component here is non-essential to the overall operator... i think @redhatHameed might have a point.

Reply:
@tchughesiv @redhatHameed My concern is that if we ignore such errors and allow the operator to continue to run, the operator may be in an unstable state (and we never tested such case).

Reply:
ah, i see ... there are dbaas controller specific issues that would warrant an exit.

Reply:
@tchughesiv right. In such situations, the operator should exit.